### PR TITLE
Migrate `run-make/crate-data-smoke` to `rmake.rs`

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @jieyouxu

### CODE REVIEW

File: tests/run-make/crate-data-smoke/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// Tests that const prop lints interrupting codegen don't leave `.o` files around.
```
Comment (lqd) [inline tests/run-make/crate-data-smoke/rmake.rs:1]: 👀

Comment (GuillaumeGomez) [reply tests/run-make/crate-data-smoke/rmake.rs]: You didn't see anything. :cat:

Comment (lqd) [reply tests/run-make/crate-data-smoke/rmake.rs]: I didn't

File: tests/run-make/crate-data-smoke/rmake.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        rustc()
+            .print("file-names")
+            .arg("--test")
+            .input("lib.rs")
+            .run(),
+        bin_name("mylib"),
+    );
+    compare_stdout(
+        rustc().print("file-names").input("lib.rs").run(),
+        rust_lib("mylib").file_name().unwrap().to_string_lossy(),
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply tests/run-make/crate-data-smoke/rmake.rs:37]: That would be much better indeed! Doing it.

File: tests/run-make/crate-data-smoke/rmake.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        rustc()
+            .print("file-names")
+            .arg("--test")
+            .input("lib.rs")
+            .run(),
+        bin_name("mylib"),
+    );
+    compare_stdout(
+        rustc().print("file-names").input("lib.rs").run(),
+        rust_lib_name("mylib"),
```
Comment (jieyouxu) [inline tests/run-make/crate-data-smoke/rmake.rs:37]: Yes, I like how this reads much better :3

**Review by jieyouxu — APPROVED** [review]

**Comment by jieyouxu** [comment]
Thanks, r=me after CI is green

**Pull request merged**